src/TokenManagers/PlayfabTokenManager.js

```diff
@@ -0,0 +1,69 @@
+const debug = require('debug')('prismarine-auth')
+const fetch = require('node-fetch')
+
+const { Endpoints } = require('../common/Constants')
+
+class PlayfabTokenManager {
+  constructor (cache) {
+    this.cache = cache
+  }
+
+  async setCachedAccessToken (data) {
+    await this.cache.setCachedPartial(data)
+  }
+
+  async getCachedAccessToken () {
+    const { pfb: cache } = await this.cache.getCached()
+
+    debug('[pf] token cache', cache)
+
+    if (!cache) return
+
+    const expires = new Date(cache.EntityToken.TokenExpiration)
+
+    const remaining = expires - Date.now()
+
+    const valid = remaining > 1000
+
+    return { valid, until: expires, data: cache }
+  }
+
+  async getAccessToken (xsts) {
+    const response = await fetch(Endpoints.PlayfabLoginWithXbox, {
+      method: 'post',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        CreateAccount: true,
```

Comment:
Same discussion as above, should we expose overrides to these options or leave it?